exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TimeProvider.java

```diff
@@ -27,4 +27,9 @@
    * Returns the current time of this time provider in UTC time zone.
    */
   ZonedDateTime getTime();
+
+  /**
+   * Returns a copy of this time provider.
```

Comment:
I do not understand how it is supposed to work. It is apparently required when testkit is
`static` **and** it has a time provider set. In this case the time provider needs to be pre-configured
in a static context (= before `@BeforeAll`). 

Some issues with the current design:
1. Does not make sense with FakeTimeProvider unless it is constant (= never changes the current time),
because the client cannot access the actual time provider used by the testkit. We can probably
make testkit return it, but I think it will only complicate things (pass one provider to the 
builder, but get another one from the teskit). On the other hand, if we do not copy (in a `static` context),
and the client puts it in a `static` field and then reconfigures in tests —
they will end up with broken initial state in case of single-threaded execution of tests or concurrency
issues (and anything broken) in case of multi-threaded execution.
2. Cannot use mocks even without the extension because they cannot be cloned (well, can with a wrapper with a throwing copy implementation).

Unless I'm missing something, copying seems restrictive and not that easy to use or fix.
But it is required only for `static` extensions. What we could do:
1. 'Fix' by adding `Testkit#getTimeProvider` and some obscure documentation.
2. Ignore the concurrency issue (i.e., `static` extension with a shared template builder 
and a shared time provider will use the same instance in all tests). If a client breaks
it by modifying the provider in any test, it is their problem.
3. Forbid `static` extensions — we seem to have found a good reason :-)
4. Anything else?

Reply:
I'd forbid (discourage?) `static` extensions. It's easier for us, will prevent any other possible issues with having a static extensions (I can't think of any right now though) and won't limit user in any way.

Reply:
Also I suppose it's better to make a copy of `templateTestKitBuilder` on every parameter resolve anyway, even though we discourage having a `static` extension. Even if someone uses `static`, they are unlikely to have any issues in most cases.

Reply:
OK, I agree with discouraging it because apparently there is no reasonable way to support `static`. Shall we also "forbid" (probably, by supporting class-level callbacks, and, if they get invoked — throwing an Exception)? Are there any downsides in doing that (i.e., is it possible that a user is forced to use `static` but it won't cause any problems)?